azure/services/aksextensions/spec.go

```diff
@@ -70,11 +70,14 @@ func (s *AKSExtensionSpec) Parameters(ctx context.Context, existingAKSExtension
 	aksExtension.Spec.Owner = &genruntime.ArbitraryOwnerReference{
 		ARMID: s.Owner,
 	}
-	aksExtension.Spec.Plan = &asokubernetesconfigurationv1.Plan{
-		Name:      ptr.To(s.Plan.Name),
-		Product:   ptr.To(s.Plan.Product),
-		Publisher: ptr.To(s.Plan.Publisher),
-		Version:   ptr.To(s.Plan.Version),
+
+	if s.Plan != nil {
```

Comment:
Do we know what a `nil` Plan would default to in aksExtension?

Reply:
It would stay as null as explained above.

Reply:
Thanks for the prompt update, makes sense :)